client/src/rpc_config.rs

```diff
@@ -175,3 +175,19 @@ impl EncodingConfig for RpcConfirmedTransactionConfig {
         }
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum EndSlotOrCommitment {
```

Comment:
naming nit: could this be something more like `RpcConfirmedBlocksConfig`

Reply:
Certainly. I'm inclined to go with `RpcConfimedBlocksWrapper`; if we need to add additional config args in the future, we'll want to drop in an `RpcConfirmedBlocksConfig` to replace CommitmentConfig.

Reply:
sgtm. `EndSlotOrCommitment` just felt a little too generic